### ISSUE TITLE
Wish: Node favourites

### ISSUE THREAD

**Issue by josols** [DynamoDS/Dynamo#7602]
There has been some buzz on the forum as well as here in the issues section about the fuzzy search algorithm in dynamo. It doesn't always seem to return the most relevant results, it's quite "fuzzy".. 
To accomodate this I suggest implementing a "Favourites" category. Browsing the node library one could possibly have a hover over option with a "heart" showing up next to the node name, pressing it would mark it as a favourite. During search, in canvas or in library these favourites would come up first in the search results. I now see that it has been requested before also: #1209

![image](https://cloud.githubusercontent.com/assets/7080648/22930037/bf61802e-f2be-11e6-848b-6fe01e81b001.png)

**Comment by mjkkirschner** [comment]
@QilongTang

**Comment by eibre** [comment]
Yes! Could it also be possible to make the search learn over time and bring the most used nodes to the top of the search results?

In the right click search it would be nice to have multi selection. Maybe with Ctrl+click. Some nodes are like siamese twins, Category and All Elements Of Category for instance.

Maybe the favorite nodes could be accessed with key board short cuts? https://github.com/DynamoDS/Dynamo/issues/4632

Speaking of fuzzy search:
![allelementsofcategory](https://cloud.githubusercontent.com/assets/5968613/22945462/9c340758-f2f4-11e6-85a2-4493bb7f6ba6.gif)

**Comment by wynged** [comment]
I've lived that GIF, I've had nightmares about that GIF.

I'd like to lend my emphatic support to a search that learns over time from
your use, and placing nodes with keyboard shortcuts!

**Comment by QilongTang** [comment]
Thanks @mjkkirschner  This is actually my biggest personal wish when I first joined Dynamo Dev Team :imp:  I did a prototype to add a favorite category where you can find the nodes you "pinned" before so you can easily access them independent to search. I thought about "the more intelligent" search as well though. I also implemented a button on node in canvas to pin the node so it can then be added to the favorite category. It's definitely doable but seems we were not clear about the favorite category or "heart node" ("pin node") whole work flow. I would encourage you guys think through this more and raise the description to @kronz and @Racel. Once the Dynamo team have a whole picture how this can be embedded into the user workflow, I think you guys can expect to see this anytime.

Thanks for raising this!

**Comment by jnealb** [comment]
@QilongTang @mjkkirschner The time is very close now :)
